src/App.jsx

```diff
@@ -1,35 +1,73 @@
-import { useEffect, useState } from "react";
-import { BrowserRouter, Route, Routes } from "react-router-dom";
-
+import React, { useEffect } from "react";
+import { BrowserRouter, Route, Routes, Navigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { setUser } from "./store/user/actions";
 import Header from "./components/Header/Header";
 import Login from "./components/Login/Login";
 import Registration from "./components/Registration/Registration";
 import Courses from "./components/Courses/Courses";
 import CreateCourse from "./components/CreateCourse/CreateCourse";
 import CourseInfo from "./components/CourseInfo/CourseInfo";
+import { PATH_URIS, TOKEN } from "./constants";
+import { ERROR_TEXT } from "./appStrings";
 
 function App() {
-  const [userName, setUserName] = useState("");
-  const isUserLogged = localStorage.getItem("token") != null;
+  const dispatch = useDispatch();
+  const isUserLogged = useSelector((state) => state.user.isAuth);
 
   useEffect(() => {
-    const tokenItem = JSON.parse(localStorage.getItem("token"));
+    const tokenItem = localStorage.getItem(TOKEN);
```

Comment:
At this point the token should be retrieved from Redux as per our discussions on Tuesday

Reply:
done